pkg/iplister/iplister_test.go

```diff
@@ -105,3 +106,141 @@ func TestValidateCIDRs(t *testing.T) {
 		assert.Equal(t, test.wantErr, err != nil, test.name)
 	}
 }
+
+func TestNewCachedIPLister(t *testing.T) {
+	t.Parallel()
+
+	il := New(&mockReader{}, &mockDecoder{}, WithTimeout(defaultTimeout))
+	ipl := NewCachedIPLister(il)
+	assert.Equal(t, defaultTimeout, ipl.lister.timeout)
+}
+
+func TestCachedIPListerGetIPs(t *testing.T) {
+	t.Parallel()
+
+	fakeData := "1.2.3.4/32"
+	decoderData := []string{fakeData}
+
+	mr := &mockReader{
+		ret: fakeData,
+		err: nil,
+	}
+	md := &mockDecoder{
+		ret: decoderData,
+		err: nil,
+	}
+	il := New(mr, md)
+	ipl := NewCachedIPLister(il)
+
+	// Test ip list population at creation time
+	ipl.lock.Lock()
+	assert.Equal(t, 1, md.count)
+	ipl.lock.Unlock()
+
+	// Test ip list is returned from cache
+	ips, err := ipl.GetIPs()
+	assert.NoError(t, err)
+	assert.Contains(t, ips, "1.2.3.4/32")
+
+	ipl.lock.Lock()
+	assert.Equal(t, 1, md.count)
+	ipl.lock.Unlock()
+
+	// Test ip list is returned from cache
+	ipl.lock.Lock()
+	md.ret = []string{"2.3.4.5/32"}
+	ipl.lock.Unlock()
```

Comment:
nit. there aren't concurrent `TestCachedIPListerGetIPs` running, so not sure why need to lock. Same comment for rest of the tests.

```suggestion
	md.ret = []string{"2.3.4.5/32"}
```

Reply:
if you run with race you can trigger field access races.